crates/nebula-backbone/src/application/path/mod.rs

```diff
@@ -5,19 +5,20 @@ use sea_orm::DatabaseConnection;
 
 use crate::{
     database::{OrganizationScopedTransaction, Persistable},
-    domain::secret::{self, Path, SecretService},
+    domain::secret::{self, AppliedPolicy, Path, SecretService},
 };
 
 pub(crate) struct PathData {
     pub path: String,
+    pub applied_policies: Vec<AppliedPolicy>,
 }
 
 #[async_trait]
 pub(crate) trait PathUseCase {
     async fn get_all(&self) -> Result<Vec<PathData>>;
-    async fn register(&self, path: &str) -> Result<()>;
+    async fn register(&self, path: &str, policies: Vec<AppliedPolicy>) -> Result<()>;
```

Comment:
I suggest changing the interface from `Vec<AppliedPolicy>` to `&[AppliedPolicy]` because using slice reference (`&[T]`) is more flexible as it allows accepting both vectors and slices as arguments